pp/go/relabeler/interface.go

```diff
@@ -40,6 +47,34 @@ type Wal interface {
 	Flush() error
 }
 
+type UnloadedDataSnapshotHeader struct {
+	Crc32        uint32
+	SnapshotSize uint32
+}
+
+func NewUnloadedDataSnapshotHeader(snapshot []byte) UnloadedDataSnapshotHeader {
+	return UnloadedDataSnapshotHeader{Crc32: crc32.ChecksumIEEE(snapshot), SnapshotSize: uint32(len(snapshot))}
+}
+
+func (h UnloadedDataSnapshotHeader) IsValid(snapshot []byte) bool {
+	return h.Crc32 == crc32.ChecksumIEEE(snapshot)
+}
+
+type UnloadedDataStorage interface {
+	WriteSnapshot(snapshot []byte) (UnloadedDataSnapshotHeader, error)
+	WriteIndex(UnloadedDataSnapshotHeader)
+	ForEachSnapshot(f func(snapshot []byte, isLast bool)) error
```

Comment:
Maybe using Range interface is better:
```suggestion
	Snapshots(f func(read func() ([]byte, error), isLast bool) bool)
```

It may be used later in manner
```go
for get := range storage.Snapshots {
	snapshot, err := get()
	if err != nil {
		return ...
	}
	// do something with snapshot
}
```

What you think?

Reply:
Code readability degradates after using range over function